topo/node/cisco/cisco.go

```diff
@@ -511,6 +512,62 @@ func (n *Node) ResetCfg(ctx context.Context) error {
 	return resp.Failed
 }
 
+// processConfig removes end command from config
+// since running it can lead to interactive prompt which is not handled.
+// Also it add commits to the end of config if it is missing
+func processConfig(cfg string) string {
+	processedCfg := ""
+	lines := strings.Split(cfg, "\n")
+	lastLine := ""
+	for _, line := range lines {
+		if strings.ToLower(strings.Trim(line, " ")) == "end" {
+			continue
+		}
+		lastLine = line
+		processedCfg += line + "\n"
+	}
+	if strings.ToLower(strings.Trim(lastLine, " ")) != "commit" {
+		processedCfg += "commit\n"
+	}
+	return processedCfg
+}
+
+func (n *Node) ConfigPush(ctx context.Context, r io.Reader) error {
+	if n.Proto.Model == ModelXRD {
+		return status.Errorf(codes.Unimplemented, "reset config is not implemented for cisco xrd node")
+	}
+
+	log.Infof("%s - pushing config", n.Name())
+
+	cfg, err := io.ReadAll(r)
+	if err != nil {
+		return err
+	}
+	cfgs := string(cfg)
+	cfgs = processConfig(cfgs)
+	log.V(1).Info(cfgs)
+
+	err = n.SpawnCLIConn()
+	if err != nil {
+		return err
+	}
+	defer n.cliConn.Close()
+
+	resp, err := n.cliConn.SendConfig(cfgs)
+	if err != nil {
+		return err
+	}
+	if resp.Failed == nil {
+		log.Infof("%s - finished config push", n.Impl.Proto.Name)
+	}
+
+	return resp.Failed
+}
+
+func (n *Node) GenerateSelfSigned(context.Context) error {
+	return status.Errorf(codes.Unimplemented, "certificate generation is not supported")
```

Comment:
can you add a short line about how it could be enabled (pushing the grpc stanza in the device cfg), if there is a link to cisco docs that would be great